packages/vm/src/environment.rs

```diff
@@ -260,6 +260,20 @@ impl<A: BackendApi, S: Storage, Q: Querier> Environment<A, S, Q> {
         });
     }
 
+    pub fn get_serialized_env(&self) -> Vec<u8> {
+        self.with_context_data(|context_data| match &context_data.serialized_env {
+            Some(env) => Ok(env.clone()),
```

Comment:
Do we need to clone env? How about to_vec()?

Reply:
yes. we need `env.clone()`. A clone is required to keep ownership of the `serialized_env` in the context.
serialized_env is already initialized after `to_vec()`.